client/src/tests/ProductDetails/pdrender.test.js

```diff
@@ -1,40 +1,209 @@
 import React from 'react';
-import { render, fireEvent, screen } from '@testing-library/react';
+import { render, fireEvent, screen, findByRole } from '@testing-library/react';
+
+import ExampleProducts from '../../../examples/Product-examples/exampleProducts.json';
+import ExampleProductInfo from '../../../examples/Product-examples/exampleProductInfo.json';
+import ExampleProductStyles from '../../../examples/Product-examples/exampleProductStyles.json';
 
 import ProductCartActions from '../../components/ProductDetails/ProductCartActions';
 import ProductGallery from '../../components/ProductDetails/ProductGallery';
 import ProductStyleSelector from '../../components/ProductDetails/ProductStyleSelector';
+import ProductPrice from '../../components/ProductDetails/ProductPrice';
 import ProductDetails from '../../components/ProductDetails';
+import ProductInfo from '../../components/ProductDetails/ProductInfo';
+import SocialShare from '../../components/ProductDetails/SocialShare';
+import Thumbnails from '../../components/ProductDetails/ProductGallery/Thumbnails';
+import Carousel from '../../components/ProductDetails/ProductGallery/Carousel';
+import SizeSelector from '../../components/ProductDetails/ProductCartActions/SizeSelector';
+import QuantitySelector from '../../components/ProductDetails/ProductCartActions/QuantitySelector';
+import Selector from '../../components/ProductDetails/ProductCartActions/Selector';
+
 
 describe('render product details', () => {
 
+  it('should render a generic selector', () => {
+    const options = new Array(5); 
+    const onChange = jest.fn(); 
+    render(<Selector name="a select" id="a select" label="some label" onChange={onChange}>
+      {options.map((_,i) => <option value={i}>{i}</option>)}
+    </Selector>);
+
+    const comboBox = screen.getByRole('combobox'); 
+
+    expect(comboBox).toBeTruthy();
+    expect(screen.getByLabelText('some label')).toBeTruthy(); 
+
+    fireEvent.change(comboBox, {target: {value: 2}}); 
+    expect(onChange).toHaveBeenCalledTimes(1); 
+    
+  })
+
+  it('should render sku select', () => {
+    const selectedStyle = ExampleProductStyles.results[0]; 
+    const inStockSkus = Object.entries(selectedStyle?.skus ?? {})?.filter(sku => (sku?.[1]?.quantity ?? 0) > 0);
+    render(<SizeSelector inStockSkus={inStockSkus} selectedSku={inStockSkus[0][0]} showLabel={true} />);
+
+    expect(screen.getByRole('combobox')).toBeTruthy();
+    expect(screen.getAllByRole('option').length).toBe(inStockSkus.length + 1);
+    expect(screen.getByLabelText('Please select size')).toBeTruthy(); 
+
```

Comment:
make sure the skuselect renders a combobox, renders all in stock skus, and has the proper label